bvm/ballerina-profiler/src/main/java/io/ballerina/runtime/profiler/codegen/StrandCheckAdapter.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (c) 2023, WSO2 LLC. (https://www.wso2.com) All Rights Reserved.
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.ballerina.runtime.profiler.codegen;
+
+import org.objectweb.asm.Label;
+import org.objectweb.asm.MethodVisitor;
+import org.objectweb.asm.Opcodes;
+import org.objectweb.asm.commons.AdviceAdapter;
+
+import static io.ballerina.runtime.profiler.util.Constants.GET_INSTANCE_DESCRIPTOR;
+import static io.ballerina.runtime.profiler.util.Constants.PROFILE_ANALYZER;
+import static io.ballerina.runtime.profiler.util.Constants.STRAND_CLASS;
+
+/**
+ * This class is used as the advice adapter for the Ballerina profiler.
+ * This class only manages the functions that contain the strand parameter.
+ *
+ * @since 2201.8.0
+ */
+public class StrandCheckAdapter extends AdviceAdapter {
+
+    Label tryStart = new Label();
+    int load;
+
+    /**
+     *
+     * @param access        access flag of the method that is wrapped
```

Comment:
Do we need method java doc here. this is not an api. also we haven't add in other classes.

Reply:
Remove the Javadoc comments. Kept some normal comments as they might give information about the implementation.